File: install/setpath.jl
# Compatibility shim for legacy pdeapps that include() this file after
# computing cdir/ii. The frontend now lives in the Exasim.jl package
# (frontends/Julia/Exasim); new code should `using Exasim` directly.
let srcdir = cdir[1:ii[end]] * "/frontends/Julia"
    if !(srcdir in LOAD_PATH)
        push!(LOAD_PATH, srcdir)
    end
end

import Exasim
# Re-export the historical bare module names for un-migrated pdeapps.
const Preprocessing = Exasim.Preprocessing
const Postprocessing = Exasim.Postprocessing
const Gencode = Exasim.Gencode
const Mesh = Exasim.Mesh
# Legacy pdeapps call Main.cmakecompile via Postprocessing.exasim; keep a
# top-level alias for any script that calls it directly.
const cmakecompile = Exasim.Gencode.cmakecompile

# Make external tools in common nonstandard prefixes visible without
# clobbering the user's PATH.
ENV["PATH"] = ENV["PATH"] * ":/usr/local/bin:/opt/local/bin:/opt/homebrew/bin";

print("==> Exasim Julia frontend (Exasim.jl package) ...\n");

File: install/setpath.py
# Compatibility shim for legacy pdeapps that exec() this file after computing
# cdir/ii. The frontend now lives in the `exasim` package
# (frontends/Python/exasim); new code should simply `import exasim`.
import sys, os

_srcdir = cdir[0:(ii+6)] + "/frontends/Python"
if _srcdir not in sys.path:
    sys.path.insert(0, _srcdir)
sys.path.append(cdir)

import exasim
# Re-export the historical bare module names for un-migrated pdeapps.
Preprocessing = exasim.Preprocessing
Postprocessing = exasim.Postprocessing
Gencode = exasim.Gencode
Mesh = exasim.Mesh
sys.modules.setdefault('Preprocessing', exasim.Preprocessing)
sys.modules.setdefault('Postprocessing', exasim.Postprocessing)
sys.modules.setdefault('Gencode', exasim.Gencode)
sys.modules.setdefault('Mesh', exasim.Mesh)

print('==> Exasim Python frontend (exasim package) ...\n')
